### PR TITLE
feat: add breakpoint thingy to docs site

### PR DESCRIPTION
**Pull request by artmsilva**
## What I did

Test link https://wpds-ui-kit-git-boooop.preview.now.washingtonpost.com/foundations/breakpoints

### CODE REVIEW

File: build.washingtonpost.com/components/Markdown/Examples/Breakpoints.jsx [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+  backgroundImage: `linear-gradient(0deg, transparent 0 50%, ${theme.colors.cta} 50% calc(50% + 2px), transparent calc(50% + 2px));`,
+  fontFamily: "monospace",
+  fontSize: theme.fontSizes["075"],
+  paddingBlockEnd: theme.space["100"],
+  color: theme.colors.cta,
+  fontWeight: theme.fontWeights.bold,
+});
+
+const Body = styled("div", {
+  paddingBlock: "1px",
```
Comment (val-l-hosler) [inline build.washingtonpost.com/components/Markdown/Examples/Breakpoints.jsx:21]: TIL ooo

File: build.washingtonpost.com/components/Markdown/Examples/Breakpoints.jsx [hunk 2]
```diff
@@ -0,0 +1,179 @@
+import React, { useState, useEffect } from "react";
+import { theme, styled, Box, RadioButton, RadioGroup } from "@washingtonpost/wpds-ui-kit";
+
+const Ruler = styled("div", {
+  zIndex: "$offer",
+  position: "fixed",
+  top: "3rem",
+  left: 0,
+  borderInline: `2px solid ${theme.colors.cta}`,
```
Comment (val-l-hosler) [inline build.washingtonpost.com/components/Markdown/Examples/Breakpoints.jsx:9]: TIL ooo

File: build.washingtonpost.com/components/Markdown/Examples/Breakpoints.jsx [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+
+const Body = styled("div", {
+  paddingBlock: "1px",
+  fontFamily: theme.fonts.meta,
+  placeItems: "left"
+});
+
+const PointsList = styled("ul", {
+  display: "flex",
+  paddingInlineStart: 0,
```
Comment (val-l-hosler) [inline build.washingtonpost.com/components/Markdown/Examples/Breakpoints.jsx:29]: TIL ooo

File: build.washingtonpost.com/components/Markdown/Examples/Breakpoints.jsx [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+  color: theme.colors.accessible,
+  fontWeight: theme.fontWeights.regular,
+});
+
+const Template = () => {
+  const [windowWidth, setWindowWidth] = useState(0);
+
+  useEffect(() => {
+    setWindowWidth(window.innerWidth);
+    const handleResize = () => setWindowWidth(window.innerWidth);
```
Comment (val-l-hosler) [inline build.washingtonpost.com/components/Markdown/Examples/Breakpoints.jsx:71]: One q, why are you setting the state with window.innerWidth, then you are setting the same state with the same value in the handleResize function?

File: build.washingtonpost.com/components/Markdown/Examples/Breakpoints.jsx [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+    window.addEventListener("resize", handleResize);
+    return () => window.removeEventListener("resize", handleResize);
```
Comment (val-l-hosler) [inline build.washingtonpost.com/components/Markdown/Examples/Breakpoints.jsx:73]: Then used here

File: build.washingtonpost.com/pages/components/[slug].js [hunk 6]
```diff
@@ -122,10 +121,26 @@ export const getStaticProps = async ({ params }) => {
...
+    } = components;
+
+    let componentList = docs.map((component) => {
+      return component.slug.replace("/components/", "");
+    });
+
+    return componentList;
+  }); 
+
+  const slug = componentsLookUptable[params.slug];
```
Comment (val-l-hosler) [inline build.washingtonpost.com/pages/components/[slug].js:138]: Can I get a TLDR of what's happening here? I'm curious.

**Review by val-l-hosler — APPROVED** [review]

File: build.washingtonpost.com/pages/components/[slug].js (hunk 6 above)
Comment (artmsilva) [reply build.washingtonpost.com/pages/components/[slug].js:138]: It's doing a safe lookup to prevent a security issue where we take unfiltered input

**Pull request merged**